### ISSUE TITLE
Fuzzy search slow with large data

### ISSUE THREAD

**Issue by s-nizamani** [Loilo/Fuse#8]
Hello,
We have a very large array with around 10 thousand records, each having following fields,
array(
    array('name' => 'name1', 'email' => 'name@gmail.com'),
    array('name' => 'second name', 'email' => 'second_email@gmail.com')
)
When searching this user's array for keys 'name' or 'email' with following settings Fuse is really slow it takes around 30, 40 seconds to find all records, I have following settings,

// initialize fuse
$settings['keys'] = array("name", "email");
$settings['distance'] = 50;
$settings['maxPatternLength'] = 16;
$settings['threshold'] = 0.6;
$fuse = new Fuse($usersArray, $settings);

// search user
$usersFound = $fuse->search("my name");

Is there a way or any setting which will help me make Fuse work faster? I tried same with JS version of Fuse and it was not that much slower took about few seconds to search for 10 thousand records

**Comment by Loilo** [comment]
I'm not deep enough into the algorithm's science to tell you about that. Fuse is literally just a port of Fuse.js, with basically literally translated source code.

The only explanation I could imagine for your problem is that the PHP engine may generally be slower than JS engines (at least regarding the algorithms used). I can't really imagine the discrepancy being as big as you described, but I'm in no way a performance expert.

In short: Sorry, I cannot help you here. If fuzzy search is important for your case and you have the technical possibility, you may try using the [V8js extension for PHP](http://php.net/manual/book.v8js.php) to run Fuse.js directly inside PHP.

**Comment by roland-deschains** [comment]
Hey @s-nizamani any updates on how to increase Fuse speed? Running really slow for me as well (using React & Express)

**Comment by loilo** [comment]
@roland-deschains How do you use the PHP version of Fuse with React & Express? Are you sure you're not using [Fuse.js](https://fusejs.io)?

I have taken a deeper look into Fuse's performance in #18, but it basically only tells that the PHP engine is kind of slow compared to V8 and therefore Fuse's overall performance is just not going to reach Fuse.js levels.
